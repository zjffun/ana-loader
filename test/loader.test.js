import {
  compile,
  execute,
  getCompiler,
  getErrors,
  getWarnings,
  readAsset,
} from './helpers';

describe('loader', () => {
  it('should work', async () => {
    const compiler = getCompiler('simple.js');
    const stats = await compile(compiler);

    const { modules } = stats.toJson();

    const result = modules.map((m) => {
      return {
        id: m.id,
        reasons: m.reasons.map((r) => r.moduleId),
      };
    });

    expect(result).toMatchSnapshot('result');

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('main.bundle.js');
    expect(getErrors(stats)).toMatchSnapshot('errors');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
  });
});
