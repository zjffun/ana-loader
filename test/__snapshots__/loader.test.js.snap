// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader should work: errors 1`] = `[]`;

exports[`loader should work: main.bundle.js 1`] = `undefined`;

exports[`loader should work: result 1`] = `
[
  {
    "id": "./simple.js",
    "reasons": [
      null,
    ],
  },
  {
    "id": "./commonjs.js",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.js",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.css",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.ts",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.vue",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.tsx",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.scss",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.less",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./dynamic-import.js",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./node_modules/test.js",
    "reasons": [
      "./foo.js",
      "./foo.ts",
      "./foo.tsx",
      "./foo.vue",
    ],
  },
  {
    "id": "./fooDep.js",
    "reasons": [
      "./foo.js",
      "./foo.ts",
      "./foo.tsx",
      "./foo.vue",
    ],
  },
  {
    "id": "./assets/test-font.woff",
    "reasons": [
      "./foo.css",
    ],
  },
  {
    "id": "./assets/test-font.ttf",
    "reasons": [
      "./foo.css",
    ],
  },
  {
    "id": "./assets/foo.png",
    "reasons": [
      "./foo.css",
      "./foo.less",
      "./foo.scss",
      "./foo.vue",
    ],
  },
  {
    "id": "./assets/foo1.png",
    "reasons": [
      "./foo.vue",
    ],
  },
  {
    "id": "./assets/foo2.png",
    "reasons": [
      "./foo.vue",
    ],
  },
  {
    "id": "./assets/foo3.png",
    "reasons": [
      "./foo.vue",
    ],
  },
  {
    "id": "",
    "reasons": [],
  },
  {
    "id": "",
    "reasons": [],
  },
  {
    "id": "",
    "reasons": [],
  },
  {
    "id": "",
    "reasons": [],
  },
]
`;

exports[`loader should work: warnings 1`] = `[]`;
